include/jsoncons/json.hpp

```diff
@@ -325,11 +325,12 @@ class basic_json
             typedef typename std::allocator_traits<Allocator>:: template rebind_alloc<string_storage_type> string_holder_allocator_type;
             typedef typename std::allocator_traits<string_holder_allocator_type>::pointer pointer;
 
-            pointer ptr_;
+            pointer ptr_ = nullptr;
 
             template <typename... Args>
             void create(string_holder_allocator_type allocator, Args&& ... args)
             {
+                JSONCONS_ASSERT(ptr_ == nullptr);
```

Comment:
I added an assert and 0 initialized variant pointers. This should add a bit more confidence that the setup is correct.